kubernetes/check.py

```diff
@@ -155,7 +150,28 @@ def _perform_kubelet_checks(self, url, instance):
             else:
                 self.service_check(service_check_base, AgentCheck.CRITICAL, tags=instance.get('tags', []))
 
+    def _configure_event_collection(self, instance):
+        self._collect_events = self.kubeutil.is_leader or _is_affirmative(instance.get('collect_events', DEFAULT_COLLECT_EVENTS))
+        if self._collect_events:
+            if self.event_retriever:
+                self.event_retriever.set_kinds(None)
+            else:
+                self.event_retriever = self.kubeutil.get_event_retriever()
+        elif self.kubeutil.collect_service_tag:
+            # Only fetch service and pod events for service mapping
+            event_delay = instance.get('service_tag_update_freq', DEFAULT_SERVICE_EVENT_FREQ)
+            if self.event_retriever:
+                self.event_retriever.set_kinds(['Service', 'Pod'])
```

Comment:
we'll need to set the delay too, via a new method to be added, and remove it when we come leader